core/src/main/java/jenkins/agents/CloudSet.java

```diff
@@ -248,6 +253,24 @@ public synchronized void doDoCreate(StaplerRequest req, StaplerResponse rsp,
         rsp.sendRedirect2(".");
     }
 
+    @POST
+    public void doReorder(StaplerRequest req, StaplerResponse rsp) throws IOException {
```

Comment:
Missing permission check allows users with SystemRead permission to reorder clouds.